LEGO1/mxdirectdraw.cpp

```diff
@@ -1,22 +1,1223 @@
+      if (result != DD_OK)
+      {
+        Error("Restore of text surface 2 failed", result);
+        return FALSE;
+      }
+    }
+  }
+
+  return TRUE;
+}
+
+// OFFSET: LEGO1 0x1009D700
+BOOL MxDirectDraw::SetPaletteEntries(
+  const PALETTEENTRY* pPaletteEntries,
+  int paletteEntryCount,
+  BOOL fullscreen)
+{
+  int reservedLowEntryCount = 10;
+  int reservedHighEntryCount = 10;
+  int arraySize = _countof(m_paletteEntries);
+  HDC hdc;
+  int i;
+
+  if (g_is_PALETTEINDEXED8)
+  {
+    hdc = GetDC(NULL);
+    GetSystemPaletteEntries(hdc, 0, arraySize, m_paletteEntries);
+    ReleaseDC(NULL, hdc);
+  }
+
+  for (i = 0; i < reservedLowEntryCount; i++)
+  {
+    m_paletteEntries[i].peFlags = 0x80;
+  }
+
+  for (i = reservedLowEntryCount; i < 142; i++)
+  {
+    m_paletteEntries[i].peFlags = 0x44;
+  }
+
+  for (i = 142; i < arraySize - reservedHighEntryCount; i++)
+  {
+    m_paletteEntries[i].peFlags = 0x84;
+  }
+
+  for (i = arraySize - reservedHighEntryCount; i < arraySize; i++)
+  {
+    m_paletteEntries[i].peFlags = 0x80;
+  }
+
+  if (paletteEntryCount != 0)
+  {
+    for (i = reservedLowEntryCount; 
+	(i < paletteEntryCount) && (i < arraySize - reservedHighEntryCount); 
+	i++)
+    {
+      m_paletteEntries[i].peRed   = pPaletteEntries[i].peRed;
+      m_paletteEntries[i].peGreen = pPaletteEntries[i].peGreen;
+      m_paletteEntries[i].peBlue  = pPaletteEntries[i].peBlue;
+    }
+  }
+  
+  if (m_pPalette != NULL)
+  {
+	HRESULT result;
+    result = m_pPalette->SetEntries(0, 0, _countof(m_paletteEntries), m_paletteEntries);
+    if (result != DD_OK)
+    {
+      Error("SetEntries failed", result);
+      return FALSE;
+    }
+  }
+
+  return TRUE;
+}
+
+// OFFSET: LEGO1 0x1009E110
+BOOL MxDirectDraw::TextToTextSurface(
+  const char* text,
+  IDirectDrawSurface* pSurface,
+  SIZE& textSizeOnSurface)
+{
+  HRESULT result;
+  HDC hdc;
+  RECT rc;
+  size_t  textLength;
+
+  if (pSurface == NULL)
+  {
+    return FALSE;
+  }
+
+  result = pSurface->GetDC(&hdc);
+  if (result != DD_OK)
+  {
+    Error("GetDC for text surface failed", result);
+    return FALSE;
+  }
+
+  textLength = strlen(text);
+  
+  SelectObject(hdc, m_hFont);
+  SetTextColor(hdc, RGB(255, 255, 0));
+  SetBkColor(hdc, RGB(0, 0, 0));
+  SetBkMode(hdc, OPAQUE);
+  GetTextExtentPoint32(hdc, text, textLength, &textSizeOnSurface);
+  SetRect(&rc, 0, 0, textSizeOnSurface.cx, textSizeOnSurface.cy);
+  ExtTextOut(hdc, 0, 0, ETO_OPAQUE, &rc, text, textLength, NULL);
+  pSurface->ReleaseDC(hdc);
+
+  return TRUE;
+}
+
+// OFFSET: LEGO1 0x1009E210
+BOOL MxDirectDraw::TextToTextSurface1(const char* text)
+{
+  return TextToTextSurface(
+    text,
+    m_pText1Surface,
+    m_text1SizeOnSurface);
+}
+
+// OFFSET: LEGO1 0x1009E230
+BOOL MxDirectDraw::TextToTextSurface2(const char* text)
+{
+  return TextToTextSurface(
+    text,
+    m_pText2Surface,
+    m_text2SizeOnSurface);
+}
+
+// OFFSET: LEGO1 0x1009E020
+void MxDirectDraw::unk1()
```

Comment:
Functions should be named after the address for less ambiguity. e.g.

```suggestion
void MxDirectDraw::FUN_1009E020()
```